library/OrleansConnector/Algorithm/InChannel.cs

```diff
@@ -104,14 +106,22 @@ internal static async Task ReceiveAsync(Guid channelId, Dispatcher dispatcher, T
                     }
                 }
             }
+            catch(HttpRequestException e) when (e.StatusCode == System.Net.HttpStatusCode.ServiceUnavailable)
+            {
+                // seems to happen occasionally
+                dispatcher.Logger.LogTrace("{dispatcher} {channelId} contact request received 503 ServiceUnavailable", dispatcher, channelId);
+            }
             catch (Exception exception)
             {
-                dispatcher.Worker.Submit(new ChannelFailedEvent()
+                if (channel.DispatcherId != null)
```

Comment:
What happens is this is `null`? Should we have a handler?

Reply:
If dispatcher is null here it means the stream died before we even found out who is on the other end. So we can of course stop listening on this stream, but we cannot tell the other side that the stream died. They should find out on their own at some point. But if we know who it is (dispatcher is not null) we can tell them immediately which may mean they can close the failed channel much quicker.

Reply:
Would be good to document this on a comment, @sebastianburckhardt, but I won't block on it